srml/contract/src/lib.rs

```diff
@@ -589,8 +589,8 @@ decl_module! {
 			}
 
 			origin_contract.storage_size -= key_values_taken.iter()
-				.map(|(_, value)| value.len() as u64)
-				.sum::<u64>();
+				.map(|(_, value)| value.len() as u32)
+				.sum::<u32>();
```

Comment:
I was looking at the last commits, seems fine.
I wonder here if we should not still use u64 (would require changing struct to use u64 with compact encoding (compact may not be a good idea). Point being that content of len more than u32 max size should never happen, but here we are using a sum of content.
Also I am not sure if when a set_value does overflow this storage_size we should fail: at this point I think we do not.

Reply:
Ahhh, I missed this...

Reply:
interesting: this content comes from child storage of account, so it is inferior to storage_size.
But there might be an issue when we actually update storage_size. storage_size is equal to actual content size + storage_size_offset. If this overflow u32 then it panics :-/

Reply:
probably better to change to u64 again for storage_size as it won't exceed 2^64 soon :-)

Reply:
https://github.com/paritytech/substrate/pull/2673
https://github.com/paritytech/substrate/issues/2672